src/spec-node/dockerCompose.ts

```diff
@@ -543,7 +543,7 @@ while sleep 1 & wait $$!; do :; done", "-"${userEntrypoint.map(a => `, ${JSON.st
     labels:${additionalLabels.map(label => `
       - '${label.replace(/\$/g, '$$$$').replace(/'/g, '\'\'')}'`).join('')}` : ''}${mounts.length ? `
     volumes:${mounts.map(m => `
-      - ${m.source}:${m.target}`).join('')}` : ''}${gpuResources}${volumeMounts.length ? `
+      - ${convertMountToVolume(m)}`).join('')}` : ''}${gpuResources}${volumeMounts.length ? `
 volumes:${volumeMounts.map(m => `
   ${m.source}:${m.external ? '\n    external: true' : ''}`).join('')}` : ''}
```

Comment:
Another case here. Not sure if anonymous volumes can be external.

Reply:
> Not sure if anonymous volumes can be external.

This should not work for anonymous volumes since we need to set the path in the `docker-compose.yml` file explicitly.

Related doc: https://docs.docker.com/compose/compose-file/07-volumes/#external

Could you please provide guidance regarding how to handle this case better? 
It seems we can fire an exception if `source` is undefined.

Reply:
Fixed in 5a202a14a4f523487b79cf9a3275ac8d748f76d7